packages/grid/x-data-grid-pro/src/hooks/features/columnResize/useGridColumnResize.tsx

```diff
@@ -556,17 +560,9 @@ export const useGridColumnResize = (
     setTimeout(() => {
       doc.removeEventListener('click', preventClick, true);
     }, 100);
-    if (columnHeaderElementRef.current) {
-      columnHeaderElementRef.current!.style.pointerEvents = 'unset';
-    }
-  }, [
-    apiRef,
-    columnHeaderElementRef,
-    handleResizeMouseMove,
-    handleResizeMouseUp,
-    handleTouchMove,
-    handleTouchEnd,
-  ]);
+    columnHeaderElementRef.current!.style.pointerEvents = 'unset';
+    columnHeaderElementRef.current = null;
+  }, [apiRef, handleResizeMouseMove, handleResizeMouseUp, handleTouchMove, handleTouchEnd]);
```

Comment:
I removed `columnHeaderElementRef` for the dependency array. It's stable, saving a bit of bundle size and runtime diff time.

Reply:
Tbh none of the dependencies here are required. `apiRef` is also a stable ref, and the rest are all stable event handlers (using `useEventCallback`). This is a limitation of react's eslint rules, and one of the reasons why I wanted something like [point 2 of the eslint rule update](https://www.notion.so/mui-org/2023-09-eslint-rules-update-eec627990f2d4a84830e9fd0981e2f6d), which wasn't implemented for lack of consensus. Besides improving the overall bundle size and runtime, it also improves the readability of the code by making it more clear what are the triggers for the effect.